var/spack/repos/builtin/packages/libfabric/package.py

```diff
@@ -99,6 +99,10 @@ class Libfabric(AutotoolsPackage):
 
     variant("debug", default=False, description="Enable debugging")
 
+    variant(
+        "uring", default=False, when="@1.17.0: fabrics=tcp", description="Enable uring support"
```

Comment:
Being a bit pendantic here, but the uring config is independent of the TCP provider, despite TCP being the only one consuming the common logic today. I would not make this conditional on the TCP fabric and leave it up to libfabric configure to do its thing.